src/udev/udev-event.c

```diff
@@ -213,6 +213,39 @@ static int get_subst_type(const char **str, FormatSubstitutionType *ret_type, ch
         return 1;
 }
 
+static int safe_atou_optional_plus(const char *s, unsigned *ret_u, bool *ret_has_plus) {
+        char *x = NULL;
+        unsigned long l;
+        bool has_plus;
+
+        assert(s);
+        assert(ret_u);
+        assert(ret_has_plus);
+
+        if (!strchr(DIGITS, s[0]))
+                return -EINVAL;
+
+        errno = 0;
+        l = strtoul(s, &x, 10);
+        if (errno > 0)
+                return -errno;
+        if (!x || x == s)
+                return -EINVAL;
+        if ((unsigned long) (unsigned) l != l)
+                return -ERANGE;
+
+        if (x[0] == '\0')
+                has_plus = false;
+        else if (x[0] == '+' && x[1] == '\0')
+                has_plus = true;
+        else
+                return -EINVAL;
+
+        *ret_u = (unsigned) l;
+        *ret_has_plus = has_plus;
+        return 0;
+}
```

Comment:
Hmm, what about the following instead:
```C
static int safe_atou_optional_plus(const char *s, unsigned *ret_u) {
    /* Returns 1 if plus, 0 if no plus, negative on error */
    int r;
    const char *p = endswith(s, "+");
    if (p)
            s = strndupa(s, p - s);
    r =  safe_atou(s, ret_u);
    if (r < 0)
           return r;
    return !!p;
}
```
(or keep `*ret_has_plus`, whatever is more convenient).